### PR TITLE
Implement ReduceSum 

### LINKED ISSUE #479: New operator for aggregation of elements in a tensor
We require a new operator to do aggregation over elements in a tensor -

For example, consider the following pytorch code to do aggregation along an axis - (https://pytorch.org/docs/stable/generated/torch.sum.html)
```
import torch

x = torch.arange(6).reshape(3,2)

print(x)

tensor([[0, 1],
        [2, 3],
        [4, 5]])

torch.sum(x, 1)

tensor([1, 5, 9])
```

We also require an operator to compute the maximum along an axis, similar to the sum reduction (https://pytorch.org/docs/stable/generated/torch.max.html)

@jiazhihao @lockshaw

### PR DESCRIPTION
**Pull request by jiazhihao**
**Description of changes:**

**Related Issues:**

Linked Issues:
- Issue # 

Issues closed by this PR:
- Closes #479 

**Before merging:**

- [ ] Did you update the [flexflow-third-party](https://github.com/flexflow/flexflow-third-party) repo, if modifying any of the Cmake files, the build configs, or the submodules?

### CODE REVIEW

**Review by gabrieleoliaro — APPROVED** [review]
LGTM! We may later want to place the reduce.cpp and reduce.cu files into the kernel folder, as part of the ongoing operators refactoring efforts, but that shouldn't affect the functioning of the operator
